dyno-core/src/main/java/com/netflix/dyno/connectionpool/Host.java

```diff
@@ -122,13 +122,18 @@ public InetSocketAddress getSocketAddress() {
         return socketAddress;
     }
 
+    /**
+     * The HostSupplier may not have the ability to return the port
+     * and therefore the hashCode() may fail if the user provides his
```

Comment:
The ```hashCode()``` doesn't 'fail', but it does cause equality checks to in fail in collections between Host objects created from the ```HostSupplier``` which may not know the dynomite port and the Host objects created by the load balancer. 

Consider rewording to state something to that effect, rather than hashCode failing.